handlerset.go

```diff
@@ -99,6 +114,21 @@ func (s HandlerSet) Filter(fn func(Handler) bool) HandlerSet {
 // RichHandlerSet is a collection of rich handlers.
 type RichHandlerSet map[Identity]RichHandler
 
+// NewRichHandlerSet returns a HandlerSet containing the given handlers.
```

Comment:
I'm not sure if this was just being generic or not.
```suggestion
// NewRichHandlerSet returns a RichHandlerSet containing the given handlers.
```